packages/suite/src/actions/wallet/send/sendFormSolanaActions.ts

```diff
@@ -82,17 +82,43 @@ const calculate = (
 // TODO(vl): For custom fees, accept a value from the user in Lamports. Then get the compute unit price by dividing it by the compute budget. See: https://docs.solana.com/proposals/fee_transaction_priority
 // Reference: https://docs.solana.com/transaction_fees
 export const composeTransaction =
-    (formValues: FormState, formState: ComposeActionContext) => () => {
+    (formValues: FormState, formState: ComposeActionContext) => async () => {
         const { account, network, feeInfo } = formState;
         const composeOutputs = getExternalComposeOutput(formValues, account, network);
-        if (!composeOutputs) return Promise.resolve(undefined); // no valid Output
+        if (!composeOutputs) return; // no valid Output
 
         const { output, decimals } = composeOutputs;
 
-        // FeeLevels are read-only
-        const { levels } = feeInfo;
+        let fetchedFee: string | undefined;
+
+        const estimatedFee = await TrezorConnect.blockchainEstimateFee({
+            coin: account.symbol,
+            request: {
+                specific: {
+                    from: account.descriptor,
+                    to:
+                        formValues.outputs.length > 0
+                            ? formValues.outputs[0].address
+                            : account.descriptor,
+                },
+            },
+        });
+
+        if (estimatedFee.success) {
+            fetchedFee = estimatedFee.payload.levels[0].feePerUnit;
```

Comment:
In general, is always a bit suspicious to access some index of array, especially like this `[0]`. I would leave a comment in code explaining here why only the first element is relevant (if that's the case)